### PR TITLE
In-band DTMF detection

### PR DESCRIPTION
**Pull request by richaas**
This PR adds in-band DTMF decoding using the Goertzel algorithm.

@alfredh please review and merge.

### CODE REVIEW

File: src/goertzel/goertzel.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Initialize goertzel state
+ *
+ * @param g     Goertzel state
+ * @param freq  Target frequency
+ * @param srate Sample rate
+ */
+void goertzel_init(struct goertzel *g, double freq, unsigned srate)
+{
```
Comment (alfredh) [inline src/goertzel/goertzel.c:23]: perhaps check the pointer to Goertzel state?

File: src/goertzel/goertzel.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+
+/**
+ * Reset goertzel state
+ *
+ * @param g Goertzel state
+ */
+void goertzel_reset(struct goertzel *g)
+{
```
Comment (alfredh) [inline src/goertzel/goertzel.c:36]: perhaps check the pointer to Goertzel state?

File: src/goertzel/goertzel.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Calculate result and reset state
+ *
+ * @param g Goertzel state
+ *
+ * @return Result value
+ */
+double goertzel_result(struct goertzel *g)
+{
```
Comment (alfredh) [inline src/goertzel/goertzel.c:50]: perhaps check the pointer to Goertzel state?

**Comment by alfredh** [comment]
thanks for this nice and useful patch. The code looks nice and clean.

The code is rather advanced with a lot of mathematical operations, so it is difficult
to review the whole code without knowing details about that specific domain.

I have tested the code with the output from `autone_dtmf` and the DTMF detection
works.

please add the new modules to README.md , some minor comments inline.


/Alfred

File: include/rem_dtmf.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+struct dtmf_dec;
+
+/**
+ * Defines the DTMF decode handler
+ *
+ * @param digit Decoded DTMF digit
+ * @param arg   Handler argument
+ */
+typedef void (dtmf_dec_h)(char digit, void *arg);
```
Comment (alfredh) [inline include/rem_dtmf.h:15]: the callback API does not support detecting the "end" of a digit, or the duration of a digit.

is this useful to add now ?

File: src/goertzel/goertzel.c (hunk 1 above)
Comment (richaas) [reply src/goertzel/goertzel.c:23]: This was avoided on purpose for performance reasons. The goertzel state struct is public and typically not dynamically allocated on its own, but part of another structure, and will therefor not likely be NULL.

File: include/rem_dtmf.h (hunk 4 above)
Comment (richaas) [reply include/rem_dtmf.h:15]: IMO this is not needed. End can be communicated as digit=0, if ever needed in the future within the current API.

**Comment by richaas** [comment]
README.md updated

**Pull request merged**